web-app/js/portal/ui/FeatureInfoPopup.js

```diff
@@ -277,22 +277,20 @@ Portal.ui.FeatureInfoPopup = Ext.extend(GeoExt.Popup, {
     _updatePopupDepthStatus: function(response) {
         if (response && response.responseXML) {
             var xmldoc = response.responseXML;
-
-            // Depth service can return 204 but our app changes that to a 200 and pipes down nothing
-            if (xmldoc && xmldoc.getElementsByTagName('depth') !== undefined) {
-                var depth = xmldoc.getElementsByTagName('depth')[0].firstChild.nodeValue;
-                var label = OpenLayers.i18n(depth <= 0 ? 'depthLabel' : 'elevationLabel');
-
-                this.depthInfoPanel.update(this.locationString + " " + this._boldify(label) + " " + Math.abs(depth) + "m");
-            }
-            else {
-                this.depthInfoPanel.update("");
+            if (xmldoc) {
+                var depthTag = xmldoc.getElementsByTagName('depth')[0].firstChild;
+                // Depth service can return 204 but our app changes that to a 200 and pipes down nothing
+                if (depthTag != null) {
+                    var depth = depthTag.nodeValue;
+                    if (depth) {
+                        var label = OpenLayers.i18n(depth <= 0 ? 'depthLabel' : 'elevationLabel');
+                        this.depthInfoPanel.update(this.locationString + " " + this._boldify(label) + " " + Math.abs(depth) + "m");
+                        return;
```

Comment:
How about:
```
var html = "&nbsp";

if (xml) {
   SOME_LOGIC;
   html = "NEW_DEPTH";
}

this.depthInfoPanel.update(html);
```

No more complex `if`s. There's just a default value can or cannot be updated.